components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ExportUtils.java

```diff
@@ -894,6 +901,33 @@ public static void addAPIMetaInformationToArchive(String archivePath, APIDTO api
                 }
             } else {
                 String asyncApiJson = RestApiCommonUtil.retrieveAsyncAPIDefinition(api, apiProvider);
+                // fetching the callback URL from asyncAPI definition.
+                JsonParser jsonParser = new JsonParser();
+                JsonObject parsedObject = jsonParser.parse(asyncApiJson).getAsJsonObject();
+                String callBackEndpoint = parsedObject.get(ASYNC_DEFAULT_SUBSCRIBER).getAsString();
```

Comment:
add a NPE for `parsedObject.get(ASYNC_DEFAULT_SUBSCRIBER)`. This would throw a NPE if `ASYNC_DEFAULT_SUBSCRIBER` is not define.

Reply:
+1 fixed